compaction.go

```diff
@@ -1,16 +1,55 @@
 package lsm
 
 func (db *DB) compactLevel(level int) error {
-	if level == 0 {
+	if level == 0 && db.needL0Compaction() {
 		return db.compactLevel0()
+	} else if level > 0 && db.needLevelNCompaction(level) {
+		return db.compactLevelN(level)
 	}
-	return db.compactLevelN(level)
+	return nil
 }
 
 func (db *DB) compactLevel0() error {
+	iterators0 := make([]*SSTableIterator, 0)
```

Comment:
### Optimizations
Consider optimizing the creation of iterators by reusing existing iterators instead of creating new ones. This can reduce memory allocation and improve performance. Additionally, ensure that the iterators are properly closed after use to avoid memory leaks.